.github/docs/how-to-version-content-on-our-website.md

```diff
@@ -0,0 +1,145 @@
+
+The website is hosted on GitHub Pages and uses a multi-version setup to allow users to switch between different versions of the documentation. Due to the nature of static websites, all versions that should later be shown on the website need to be built in parallel and deployed to the same `public/` directory, with each version living in its own subfolder.
+
+## ⚙️ Requirements
+
+- Node.js >=22.12.0
+- npm >=10.9.0
+- Hugo (will be installed via npm)
+
+To be able to test your changes with a locally running Hugo server, install EXACTLY the dependencies that are defined for the current version (npm is using the file `package-lock.json`):
+
+```sh
+npm ci
+```
+
+## The Basic Setup of the Multi-Version Website in a Nutshell
+
+- **Version branches** e.g., `website/vx.y` for each released website version.
+
+- A **central file `data/versions.json`** to power the version switcher. This file contains all versions to be used on the website. We'll have these versions:
+
+  - `current` holds the content of our current website, dedicated to OCM v1.
+  
+  - `dev` holds the content dedicated to the ongoing development in the `main` branch of the `open-component-model` monorepo, dedicated to OCM v2.
+  
+  - `vx.y` hold the content of the released versions of the website, e.g., `v1.0`, `v2.0`, etc.
+
+- **Hugo Modules** to include reference documentation for the CLI and APIs from other repositories. The modules are created from the Go submodules in the remote Git repositories using tags, e.g. `ocm.software/open-component-model/cli` for the OCM CLI. The modules (or subfolder inside them) are then mounted into the website. The configuration for these modules is stored in `config/_default/module.toml` and MUST only be maintained in the `main` branch.
+
+- A **build output directory `public/`** that is created during each build of the website and contains all versions in parallel in different subfolders. The default version displayed when the website is opened, most-likely the latest released version, lives in the root `public` folder, other versions in subfolders like `public/v1.0`. This is required for the static website hosted as GitHub Pages deployment to work correctly, as the complete website is deployed from the `public/` directory. Only then the version switcher works correctly and offers back-and-forth navigation.
+
+- Using **URL patterns** like <https://ocm.software>`/vx.y/`> to access the versioned content, where `vx.y` is the version tag.
+
+## 🛠️ Versioning Process Step-by-Step
+
+1. 🧩 **Define Modules for Reference Docs (in `main`)**
+   - Configure Hugo modules in `config/_default/module.toml`.
+
+   - Example:
+
+     ```toml
+     [[imports]]
+     path = "ocm.software/open-component-model/cli"
+       [[imports.mounts]]
+         source = "docs/reference"
+         target = "content/docs/reference/ocm-cli"
+     ```
+
+2. 🎯 **Set Default Version (in `main`)**
```

Comment:
formatting here is off, thats why the diff preview is red

Reply:
In the preview and later rendered doc it looks fine. The diff maybe shows that due to the behavior of markdown for the mix of sorted and unsorted lists. I use a sorted list for the process steps and below an unordered list for the points belonging to each process step. If I make the unordered list a level one list, then the ordered list breaks, so I have to decide for the minor issue. And that for me is seeing level two indents for the unordered list, but can keep the ordered lists working.

Can you please check in the Preview if it looks ok there? The current doc does not show any markdown linter errors.